public/javascripts/study/tour-chapter.js

```diff
@@ -87,5 +87,11 @@ lichess.studyTourChapter = function (study) {
       tour.addStep(s.title, s);
     });
     tour.start();
+
+    lichess.pubsub.on('tour.stop', tour.cancel);
+
+    Shepherd.once('inactive', _ => {
+      lichess.pubsub.off('tour.stop', tour.cancel);
+    });
```

Comment:
`inactive` covers all ways the tour ends (eg. user completes it, user cancels it, we cancel it programmatically)
`once` makes sure this listener only sticks around for the lifespan of the current tour